table/block_based/block_based_table_iterator.h

```diff
@@ -329,6 +329,9 @@ class BlockBasedTableIterator : public InternalIteratorBase<Slice> {
   // is used to disable the lookup.
   IterDirection direction_ = IterDirection::kForward;
 
+  // Saves the key after each operation.
+  Slice last_key_;
```

Comment:
Couldn't we get this from the block iter when needed? I'm worried that this might become stale.

Reply:
> Couldn't we get this from the block iter when needed? I'm worried that this might become stale.
Yeah! That's actually works better.